checker/src/org/checkerframework/checker/units/UnitsAnnotatedTypeLoader.java

```diff
@@ -0,0 +1,49 @@
+package org.checkerframework.checker.units;
+
+import org.checkerframework.checker.units.qual.UnitsMultiple;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.AnnotatedTypeLoader;
+import org.checkerframework.framework.util.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.lang.annotation.Annotation;
+
+import javax.lang.model.element.AnnotationMirror;
+
+public class UnitsAnnotatedTypeLoader extends AnnotatedTypeLoader {
+
+    public UnitsAnnotatedTypeLoader(BaseTypeChecker checker) {
+        super(checker);
+    }
+
+    /**
+     * Custom filter for units annotations:
+     *
+     * This filter will ignore (by returning false) any units annotation which is an alias of another
+     * base unit annotation (identified via {@link UnitsMultiple} meta-annotation). Alias annotations
+     * can still be used in source code; they are converted into a base annotation by
+     * {@link UnitsAnnotatedTypeFactory#aliasedAnnotation(AnnotationMirror)}. This filter simply
+     * makes sure that the alias annotations themselves don't become part of the type hierarchy as
+     * their base annotations already are in the hierarchy.
+     */
+    @Override
+    protected boolean isSupportedAnnotationClass(Class<? extends Annotation> annoClass) {
+        // build the initial annotation mirror (missing prefix)
+        AnnotationBuilder builder = new AnnotationBuilder(processingEnv, annoClass);
+        AnnotationMirror initialResult = builder.build();
+
+        // further refine to see if the annotation is an alias of some other SI Unit annotation
+        for (AnnotationMirror metaAnno : initialResult.getAnnotationType().asElement().getAnnotationMirrors() ) {
```

Comment:
initialResult.getAnnotationType().asElement().getAnnotationMirrors() might throw an com.sun.tools.javac.code.Symbol.CompletionFailure.  

We had this problem elsewhere, see issue #309 which was fixed in 930a8312.   I think we should add a getMetaAnnotations(AnnotationMirror) to AnnotationUtils that that catches this exceptions.  Could you either do this, probably in a new pull request, or email me and I can do it?